File: java/src/com/rohitsh/recursion/ReverseStringII.java
package com.rohitsh.recursion;

public class ReverseStringII {

    private static void helper(char[] s, int left, int right){
        // base case
        if(left >= right) return;

        char temp = s[left];
        s[left] = s[right];
        s[right] = temp;

        left++;
        right--;

        helper(s, left, right);


    }

    public static void reverseString(char[] s) {
        helper(s, 0, s.length - 1);
    }

    public static void main(String[] args) {

        var s = new char[]{'h', 'e', 'e'};
        reverseString(s);

        System.out.println(s);

    }

}

File: c-sharp/recursion/ReverseStringII.cs
namespace c_sharp.recusrsion
{
    public class ReverseStringII
    {
        public static void Reverse(char[] s)
        {
            int left = 0, right = s.Length - 1;

            Helper(s, left, right);
        }

        private static void Helper(char[] s, int left, int right)
        {
            if(left >= right) return;

            // swap two indexes
            char temp = s[left];
            s[left] = s[right];
            s[right] = temp;

            // call helper with next indexes
            left++; right--;

            Helper(s, left, right);
        }
    }
}